modules/markup/html.go

```diff
@@ -43,7 +43,7 @@ var (
 	IssueAlphanumericPattern = regexp.MustCompile(`( |^|\()[A-Z]{1,10}-[1-9][0-9]*\b`)
 	// CrossReferenceIssueNumericPattern matches string that references a numeric issue in a different repository
 	// e.g. gogits/gogs#12345
-	CrossReferenceIssueNumericPattern = regexp.MustCompile(`( |^)[0-9a-zA-Z]+/[0-9a-zA-Z]+#[0-9]+\b`)
+	CrossReferenceIssueNumericPattern = regexp.MustCompile(`( |^)[0-9a-zA-Z-_\.]+/[0-9a-zA-Z-_\.]+#[0-9]+\b`)
```

Comment:
Before anyone researches this: did that already - both RegisterForm and CreateRepoForm in modules/auth use AlphaDashDot as a rule in the binding field, which in turn inside macaron is converted into "Every string that matches [this](https://sourcegraph.com/github.com/go-gitea/gitea/-/blob/vendor/github.com/go-macaron/binding/binding.go#L247) should be rejected".